projects/demo/src/modules/markup/icons/pipes/icon-highlight.pipe.ts

```diff
@@ -1,9 +1,17 @@
 import {Pipe, PipeTransform} from '@angular/core';
-import {TUI_DEFAULT_MATCHER} from '@taiga-ui/cdk';
+
+import {DemoTuiIcon} from '../icons.tokens';
 
 @Pipe({name: `iconHighlight`})
 export class IconHighlightPipe implements PipeTransform {
-    transform(name: string, search: string): boolean {
-        return search.length > 1 && TUI_DEFAULT_MATCHER(name, search);
+    transform(
+        icons: ReadonlyArray<DemoTuiIcon | string>,
+        search: string,
+    ): ReadonlyArray<DemoTuiIcon | string> {
+        const searchSubstring = search.trim().toLowerCase();
+
+        return searchSubstring.length > 1
+            ? icons.filter(icons => icons.toLowerCase().includes(searchSubstring))
```

Comment:
I think you can just use `tuiFilter` pipe with `TUI_DEFAULT_MATCHER`

Reply:
fixed